### PR TITLE
Establish lock to avoid race scenario in subscriptions

### PR DESCRIPTION
**Pull request by SabreCat**
We've been having an issue where, particularly with Google Play subscriptions, a number of operations done during subscription processing are happening twice. Messages sent to Slack, analytics events, and so on.

This PR adds a custom lock property, similar to our "cron signature", so that as a first step when creating a user subscription we block further user subscription ops until the first has finished.

Questions for review:

- I don't feel great about importing the User model here, but that's necessary for `findOneAndUpdate`... right? Should I include _subSignature in `user.data` and stick with `save()`?
- The problem case has thus far only been on user subs, but should I implement this for groups as well?

### CODE REVIEW

**Review by phillipthelen — APPROVED** [review]
This looks good to me.
I'm not sure importing the user model would be an issue. Why would/could it?

I think for now just using this on user subscriptions is ok.

**Comment by SabreCat** [comment]
@negue we're a little stumped on why these tests are still failing... any ideas?

File: website/server/libs/payments/subscriptions.js [hunk 1]
```diff
@@ -79,6 +80,20 @@ async function createSubscription (data) {
...
+      {
+        _id: data.user._id,
+        $or: [
+          { _subSignature: 'NOT_RUNNING' },
+          { _subSignature: { $exists: false } },
+        ],
+      },
+      { $set: { _subSignature: 'SUB_IN_PROGRESS' } },
+    );
+    if (!unlockedUser) throw new NotFound('User not found or subscription already processing.');
```
Comment (negue) [inline website/server/libs/payments/subscriptions.js]: better use `{}` so many single line if conditions we might loose the real sight of what will/should be executed when - I think I should search for a lint for it xD

**Review by negue — APPROVED** [review]
one remark to all those condition lines


as for the error - looking at ` it('errors when a user is already subscribed', ` it doesn't have a `user.save` - which might be the issue

**Comment by SabreCat** [comment]
I do not even slightly understand why we need to run a `save()` on a brand new created user but it seems to work @.@

**Comment by negue** [comment]
I think the `new User` just creates the object but not adds it to the mongo collection and then the rest doesn't can't work

**Pull request merged**